walkeeper/src/wal_service.rs

```diff
@@ -460,13 +474,12 @@ impl System {
                 match file.try_lock_exclusive() {
                     Ok(()) => {}
                     Err(e) => {
-                        panic!(
+                        io_error!(
                             "Control file {:?} is locked by some other process: {}",
                             &control_file_path, e
                         );
                     }
```

Comment:
I think we should retire `io_error!`.  `.with_context()` seems like it handles this case well.